presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/RecordingHiveMetastore.java

```diff
@@ -70,7 +71,7 @@
     private final Cache<String, Optional<List<String>>> allViewsCache;
     private final Cache<HivePartitionName, Optional<Partition>> partitionCache;
     private final Cache<HiveTableName, Optional<List<String>>> partitionNamesCache;
-    private final Cache<PartitionFilter, Optional<List<String>>> partitionNamesByPartsCache;
+    private final Cache<String, Optional<List<String>>> partitionNamesByFilterCache;
```

Comment:
You should probably just fix the serialization, and revert all the changes to this file.

Reply:
Same comment as above. Keeping it as string since PartitionFilter is not Json serializable anymore